<?xml version="1.0"?>
<launch>
  <remap from="/image" to="/camera/color/image_raw/compressed"/>
  
  <node pkg="zbar_ros" type="barcode_reader_node" name="barcode_reader_node"/>
      
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find turtlebot3_navigation)/maps/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>
 -->
 
  <!-- Map server -->
  <!-- node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/-->
  <node pkg="map_server" type="map_server" name="map_server" args="/home/slaven/catkin_ws/map/mymap.yaml"/>

  <!-- AMCL -->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch">
  	<arg name="initial_pose_x" value="-4.37"/>
  	<arg name="initial_pose_y" value="-0.94"/>
  </include>

  <!-- move_base
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>
   -->
   
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/costmap_common_params_real.yaml" command="load" ns="global_costmap"/>
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/costmap_common_params_real.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/move_base_params.yaml" command="load" />
      <rosparam file="$(find rc_simul_worlds)/params/turtlebot3/dwa_local_planner_params.yaml" command="load" />
    </node>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>
</launch>
